metaseq/data/prompt_generator.py

```diff
@@ -0,0 +1,314 @@
+from argparse import ArgumentError
+import json
+import random
+import math
+from jinja2 import Environment, meta
+
+from collections import defaultdict
+from glob import glob
+from tokenizers import ByteLevelBPETokenizer, Tokenizer
+from typing import Generator, List, Optional
+
+from metaseq.logging import get_logger
+
+logger = get_logger(__name__)
+
+prompt_template_default = """
+{%- for shot in shots -%}
+{{ shot['src'] }} {{ shot['tgt'] }}
+
+{%+ endfor -%}
+{{ input_s }}"""
+
+
+class PromptGenerator:
+    """
+    This class is responsible for decorating the current sample that is intended
+    to be used as the input to the LM with additional information. The goal of
+    adding this additional information is to help the LM better understand the
+    context of the current sample.
+
+    See :meth:`get_next_prompt` for more details on how the prompt is decorated
+    and what decorations are possible.
+    """
+
+    def __init__(
+        self,
+        prompt_template_s: Optional[str] = None,
+        n_fewshot_samples: int = 0,
+        few_shot_data_glob_path: Optional[str] = None,
+        fewshot_sample_method='random',
+        rng_seed=42,
+        max_tokens: int = None,
+        shots_to_input_truncation_percentage: float = 0.5,
+        tokenizer_vocab_file_path="/mnt/input_data_dir/pretrained_models/OPT/dependencies/gpt2-vocab.json",
+        tokenizer_merges_file_path="/mnt/input_data_dir/pretrained_models/OPT/dependencies/gpt2-merges.txt",
```

Comment:
If `Metaseq` has a standardized path for the _vocab_ and _merges_ files then we'll need to replace them here :) If not we might need to remove the default value.